bitnami/thanos/templates/query/deployment.yaml

```diff
@@ -86,21 +86,32 @@ spec:
             - --store={{ . }}
             {{- end }}
             {{- if $query.grpcTLS.server.secure }}
+            {{- if $query.grpcTLS.server.existingSecret.name }}
```

Comment:
Sorry, I may not explain well. As we used in the referred PR above, I meant to use https://github.com/bitnami/charts/tree/master/bitnami/common#existingsecret in order to make this compatible with external secrets. I think it could be good to have something like

```
{{- if or $query.grpcTLS.server.existingSecret.name $query.grpcTLS.server.secure }}
- --grpc-server-tls-cert=/tls/server/{{ include "common.secrets.key" (dict "existingSecret" .Values...existingSecret "key" "tlsCert") }}
...
...
``` 

And also do the same for the client (same logic as server). Then in the volume:

```
{{- if or $query.grpcTLS.server.secure $query.grpcTLS.server.existingSecret.name }}
        - name: tls-server
          secret:
            secretName: {{ include "common.secrets.name" (dict "existingSecret" .Values...existingSecret "context" $) }}
{{- end }}
```

Reply:
@dani8art I see. One thing that I'm not getting is what if the current secret is used (the one part of the template)?

Reply:
@dani8art I see. What about not using an existing secret and just setting up the values of ca, cert, and key, which results in a secret being created. doing  `{{ include "common.secrets.name" (dict "existingSecret" .Values...existingSecret "context" $) }}` will not result in setting the correct secret name.
Am I missing anything?

Reply:
We would like to maintain the `existingSecret` feature.

Reply:
@dani8art I don't understand

Reply:
@dani8art Do you mean we should only rely on the `existingSecret` field and remove the secrets created as part of the template?

Reply:
No, I mean you could use this common helper that was created to support both as we do in other solutions:
- helpers code ("common.secrets.name", "common.secrets.key"): https://github.com/bitnami/charts/blob/master/bitnami/common/templates/_secrets.tpl
- a generic example of use: https://github.com/bitnami/charts/tree/master/bitnami/common#existingsecret
-  a specific case: https://github.com/bitnami/charts/pull/5189

My last comment was about refactoring the `if/else` logic to not repeat some pieces of code.

Reply:
I see @dani8art. That means the current secrets' names in the templates need to change. If that's okay, I'll be happy to push a fix, its cleaner with the common helper.

Reply:
@dani8art I just pushed a draft, looks cleaner to me. Thanks for the help. Let me know what you think